File: app/schemas/employee.py
import uuid

from pydantic import BaseModel, Field

from app.models import EmployeeRole


class EmployeeCreate(BaseModel):
    username: str = Field(min_length=3, max_length=80)
    password: str = Field(min_length=6, max_length=128)
    role: EmployeeRole
    store_id: uuid.UUID | None = None


class EmployeeUpdate(BaseModel):
    username: str | None = Field(default=None, min_length=3, max_length=80)
    role: EmployeeRole | None = None
    store_id: uuid.UUID | None = None


class EmployeeResponse(BaseModel):
    id: uuid.UUID
    username: str
    role: EmployeeRole
    store_id: uuid.UUID | None
    store_name: str | None
    is_active: bool


class EmployeePasswordReset(BaseModel):
    new_password: str = Field(min_length=6, max_length=128)

File: frontend/src/services/employee.ts
import { request } from '../utils/request';

export type EmployeeRole = 'admin' | 'store_manager' | 'staff';

export interface IEmployeeRecord {
  id: string;
  username: string;
  role: EmployeeRole;
  store_id: string | null;
  store_name: string | null;
  is_active: boolean;
}

export interface CreateEmployeePayload {
  username: string;
  password: string;
  role: EmployeeRole;
  store_id: string | null;
}

export interface UpdateEmployeePayload {
  username?: string;
  role?: EmployeeRole;
  store_id?: string | null;
}

export async function getEmployees(): Promise<IEmployeeRecord[]> {
  const response = await request.get<IEmployeeRecord[]>('/api/employees');
  return response.data;
}

export async function createEmployee(payload: CreateEmployeePayload): Promise<IEmployeeRecord> {
  const response = await request.post<IEmployeeRecord>('/api/employees', payload);
  return response.data;
}

export async function updateEmployee(id: string, payload: UpdateEmployeePayload): Promise<IEmployeeRecord> {
  const response = await request.put<IEmployeeRecord>(`/api/employees/${id}`, payload);
  return response.data;
}

export async function toggleEmployeeActive(id: string): Promise<IEmployeeRecord> {
  const response = await request.put<IEmployeeRecord>(`/api/employees/${id}/toggle-active`);
  return response.data;
}

export async function resetEmployeePassword(id: string, newPassword: string): Promise<{ message: string }> {
  const response = await request.put<{ message: string }>(`/api/employees/${id}/reset-password`, {
    new_password: newPassword,
  });
  return response.data;
}
